src/DynamoCoreWpf/ViewModels/Watch3D/AttachedProperties.cs

```diff
@@ -12,6 +12,85 @@ namespace Dynamo.Wpf.ViewModels.Watch3D
     /// </summary>
     public static class AttachedProperties
     {
+
+        // handles determining color of elementGeometry3Ds when any property is set false
+        // as the state of other properties must be checked to determine the correct color / material.
+        private static void OnPropertySetFalse(DependencyObject obj)
+        {
+            if (obj is GeometryModel3D && obj.GetType() != typeof(BillboardTextModel3D))
+            {
+                //mesh case
+                var geom = (GeometryModel3D)obj;
+                var meshGeom = geom as DynamoGeometryModel3D;
+                if (meshGeom != null)
+                {
+
+                    // if selection is not enabled after this property was set false
+                    // then determine transparencey. IE selection state should always override
+                    // the other properties.
+                    if (!GetShowSelected(meshGeom))
+                    {
+                        if (GetIsolationMode(meshGeom))
+                        {
+                            meshGeom.Material = HelixWatch3DViewModel.IsolatedMaterial;
+                        }
+                        else if (GetIsFrozen(meshGeom))
+                        {
+                            meshGeom.Material = HelixWatch3DViewModel.FrozenMaterial;
+                        }
+                        else
+                        {
+                            //TODO handle vertex coloring in all these cases
+                            meshGeom.Material = HelixWatch3DViewModel.WhiteMaterial;
```

Comment:
Remind me what state it is other than selection, isolate and frozen?

Reply:
normal state.